contracts/dnsregistrar/DNSRegistrar.sol

```diff
@@ -60,35 +69,19 @@ contract DNSRegistrar is IDNSRegistrar {
         emit NewPublicSuffixList(address(suffixes));
     }
 
-    /**
-     * @dev Claims a name by proving ownership of its DNS equivalent.
-     * @param name The name to claim, in DNS wire format.
-     * @param proof A DNS RRSet proving ownership of the name. Must be verified
-     *        in the DNSSEC oracle before calling. This RRSET must contain a TXT
-     *        record for '_ens.' + name, with the value 'a=0x...'. Ownership of
-     *        the name will be transferred to the address specified in the TXT
-     *        record.
-     */
-    function claim(bytes memory name, bytes memory proof) public override {
-        (bytes32 rootNode, bytes32 labelHash, address addr) = _claim(name, proof);
-        ens.setSubnodeOwner(rootNode, labelHash, addr);
-    }
-
     /**
      * @dev Submits proofs to the DNSSEC oracle, then claims a name using those proofs.
-     * @param name The name to claim, in DNS wire format.
-     * @param input The data to be passed to the Oracle's `submitProofs` function. The last
-     *        proof must be the TXT record required by the registrar.
-     * @param proof The proof record for the first element in input.
+     * @param input A chain of signed DNS RRSETs ending with a text record.
      */
-    function proveAndClaim(bytes memory name, DNSSEC.RRSetWithSignature[] memory input, bytes memory proof) public override {
-        proof = oracle.submitRRSets(input, proof);
-        claim(name, proof);
+    function proveAndClaim(bytes memory name, DNSSEC.RRSetWithSignature[] memory input) public override {
+        bytes memory record = oracle.verifyRRSet(input);
+        (bytes32 rootNode, bytes32 labelHash, address addr) = _claim(name, record);
+        ens.setSubnodeOwner(rootNode, labelHash, addr);
     }
 
-    function proveAndClaimWithResolver(bytes memory name, DNSSEC.RRSetWithSignature[] memory input, bytes memory proof, address resolver, address addr) public override {
-        proof = oracle.submitRRSets(input, proof);
-        (bytes32 rootNode, bytes32 labelHash, address owner) = _claim(name, proof);
+    function proveAndClaimWithResolver(bytes memory name, DNSSEC.RRSetWithSignature[] memory input, address resolver, address addr) public override {
+        bytes memory record = oracle.verifyRRSet(input);
+        (bytes32 rootNode, bytes32 labelHash, address owner) = _claim(name, record);
         require(msg.sender == owner, "Only owner can call proveAndClaimWithResolver");
         if(addr != address(0)) {
             require(resolver != address(0), "Cannot set addr if resolver is not set");
```

Comment:
Shall we replace with a custom error?

Reply:
I'll tackle that as part of a future PR.